app/services/sufia/user_stat_importer.rb

```diff
@@ -46,81 +54,93 @@ def sorted_users
       end
+      ids
+    end
 
-      # For each date, add the view and download counts for this
-      # file to the view & download sub-totals for that day.
-      # The resulting hash will look something like this:
-      # {"2014-11-30 00:00:00 UTC" => {:views=>2, :downloads=>5},
-      #  "2014-12-01 00:00:00 UTC" => {:views=>4, :downloads=>4}}
-      def tally_results(file_stats, stat_name, total_stats)
-        file_stats.each do |stats|
-          # Exclude the stats from today since it will only be a partial day's worth of data
-          break if stats.date == Date.today
-
-          date_key = stats.date.to_s
-          old_count = total_stats[date_key] ? total_stats[date_key].fetch(stat_name) { 0 } : 0
-          new_count = old_count + stats.method(stat_name).call
-
-          old_values = total_stats[date_key] || {}
-          total_stats.store(date_key, old_values)
-          total_stats[date_key].store(stat_name, new_count)
-        end
-        total_stats
+    # Added 1-26-16 gordonl
+    def work_ids_for_user(user)
+      ids = []
+      ::Work.find_in_batches("#{Solrizer.solr_name('depositor', :symbol)}:\"#{user.user_key}\"", fl: "id") do |group|
+        ids.concat group.map { |doc| doc["id"] }
       end
+      ids
+    end
 
-      def create_or_update_user_stats(stats, user)
-        stats.each do |date_string, data|
-          date = Time.zone.parse(date_string)
+    # Modified 1-26-16 gordonl
+
+    # For each date, add the view and download counts for this
+    # file to the view & download sub-totals for that day.
+    # The resulting hash will look something like this:
+    # {"2014-11-30 00:00:00 UTC" => {:views=>2, :downloads=>5},
+    #  "2014-12-01 00:00:00 UTC" => {:views=>4, :downloads=>4}}
+    def tally_results(current_stats, stat_name, total_stats)
+      current_stats.each do |stats|
+        # Exclude the stats from today since it will only be a partial day's worth of data
+        break if stats.date == Date.today
+
+        date_key = stats.date.to_s
+        old_count = total_stats[date_key] ? total_stats[date_key].fetch(stat_name) { 0 } : 0
+        new_count = old_count + stats.method(stat_name).call
+
+        old_values = total_stats[date_key] || {}
+        total_stats.store(date_key, old_values)
+        total_stats[date_key].store(stat_name, new_count)
+      end
+      total_stats
+    end
 
-          user_stat = UserStat.where(user_id: user.id).where(date: date).first
-          user_stat ||= UserStat.new(user_id: user.id, date: date)
+    def create_or_update_user_stats(stats, user)
+      stats.each do |date_string, data|
+        date = Time.zone.parse(date_string)
 
-          user_stat.file_views = data[:views] || 0
-          user_stat.file_downloads = data[:downloads] || 0
-          user_stat.save!
-        end
-      end
+        user_stat = UserStat.where(user_id: user.id).where(date: date).first
+        user_stat ||= UserStat.new(user_id: user.id, date: date)
```

Comment:
Instead of doing this, can you do `user_stat = UserStat.where(...).first_or_initialize` on the previous line.